text/src/tail.rs

```diff
@@ -117,29 +116,46 @@ fn tail(args: &Args) -> Result<(), Box<dyn std::error::Error>> {
     };
     let mut reader = io::BufReader::new(file);
 
-    match args.bytes {
-        Some(bytes) => print_last_n_bytes(&mut reader, bytes)?,
-        None => print_last_n_lines(reader, args.lines.unwrap())?,
+    if let Some(bytes) = args.bytes {
+        print_last_n_bytes(&mut reader, bytes)?;
+    } else {
+        print_last_n_lines(reader, args.lines.unwrap())?;
     }
 
     // If follow option is specified, continue monitoring the file
     if args.follow {
         let file_path = args.file.as_ref().unwrap();
-        let mut file = fs::File::open(file_path)?;
+        // Initialization of inotify
+        let mut inotify = Inotify::init()?;
+        inotify.add_watch(file_path, WatchMask::MODIFY)?;
 
-        // Seek to the end of the file
+        // Opening a file and placing the cursor at the end of the file
+        let mut file = File::open(file_path)?;
         file.seek(SeekFrom::End(0))?;
         let mut reader = BufReader::new(file);
 
+        // Buffer for inotify events
+        let mut buffer = [0u8; 4096];
+
         loop {
-            let mut buffer = String::new();
-            let bytes_read = reader.read_line(&mut buffer)?;
-            if bytes_read > 0 {
-                print!("{}", buffer);
-                io::stdout().flush()?;
+            // Read inotify events
+            let events = inotify.read_events_blocking(&mut buffer)?;
+
+            // Handle each event
+            for event in events {
+                if event.mask.contains(inotify::EventMask::MODIFY) {
+                    // If the file has been modified, read the new lines and output them
+                    let mut new_data = String::new();
+                    let bytes_read = reader.read_to_string(&mut new_data)?;
```

Comment:
File could be truncated. This would require moving the cursor, but it works somehow rightnow.
Original tail informs on truncation:
`tail: <file>: file truncated`